src/loop_logic.rs

```diff
@@ -220,7 +220,22 @@ impl<'l, Data> EventLoop<'l, Data> {
     ///
     /// Fails if the initialization of the polling system failed.
     pub fn try_new() -> crate::Result<Self> {
-        let poll = Poll::new()?;
+        Self::inner_new(false)
+    }
+
+    /// Create a new event loop in high precision mode
+    ///
+    /// On some platforms it requires to setup more resources to enable high-precision
+    /// (sub millisecond) capabilities, so you should use this constructor if you need
+    /// this kind of precision.
+    ///
+    /// Fails if the initialization of the polling system failed.
+    pub fn try_new_high_precision() -> crate::Result<Self> {
```

Comment:
API question: does it make more sense to

- have this be a runtime check ie. return a `Result` or
- be feature gated, so that it doesn't even compile on systems that don't support it

...? In other words, do we expect anyone to do

```rust
let mut event_loop = match EventLoop::try_new_high_precision() {
  Ok(loop) => loop,
  Err(_) => {
    log::warn!("High precision event loop not supported! Falling back on regular old event loop.");
    EventLoop::try_new().expect("Couldn't create event loop")
  }
};
```

Or is it more likely that anyone who _needs_ the HP loop will not be willing to continue if only the regular loop is available?

Reply:
If your application is extremely sensitive to timings (namely a compositor trying to present at 360hz, you will hit a fps cap due to timer precision limitations.

Support for timerfd needs to be determined at runtime. Kqueue based polling always work in high precision.

Reply:
I don't know much about compositors - is hitting that cap something that still allows you to function?

Reply:
Yeah you could still function, depending on the context it may just be slightly less responsive than you expect or have tearing.

Reply:
Question answered then.

Reply:
> Or is it more likely that anyone who needs the HP loop will not be willing to continue if only the regular loop is available?

I think that, at least you would want to know that the HP loop could not be initialized and be aware that you're going to be running low-precision, possibly to adjust your logic accordingly (setting a flag somewhere, logging a warning, putting a cap on the screen refresh rate you'll allow the user to configure, etc...)